### ISSUE TITLE
Targeted LinfPGD Attack Behavior?

### ISSUE THREAD

**Issue by MetaMain** [bethgelab/foolbox#610]
Hi, I am having an issue with the Foolbox LinfPGD attack in PyTorch. I am doing a targeted attack on a ResNet50 model trained on CIFAR-10. Previously, I ran the attack in another software (Cleverhans/Tensorflow) and I was able to achieve a 100% attack success rate with eps=0.05 and 10 steps. However, in Foolbox I can only achieve the same performance if I run the attack for 1000 steps. Am I somehow running the attack wrong? Any guidance would be greatly appreciated.  Foolbox example code example below:

import os
import torchvision.models as models
import eagerpy as ep
from foolbox import PyTorchModel, accuracy, samples
from foolbox.attacks import LinfPGD
import foolbox 
import torch
import resnet
import DataManagerPytorch as DMP
import torchvision.datasets as datasets
import torchvision.transforms as transforms
import numpy 

def main():
    #Setup the model 
    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    model = resnet.resnet50(pretrained=True).eval().cuda()
    preprocessing = dict(mean=[0.4914, 0.4822, 0.4465], std=[0.2023, 0.1994, 0.2010], axis=-3)
    fmodel = PyTorchModel(model, bounds=(0, 1), preprocessing=preprocessing)
    #CIFAR-10
    attackSampleNum = 100
    normalize = transforms.Normalize(mean=[0.0, 0.0, 0.0], std=[1.0,1.0, 1.0])
    valLoader = torch.utils.data.DataLoader(datasets.CIFAR10(root='./data', train=False, download=True, transform=transforms.Compose([transforms.ToTensor(), normalize,])), batch_size=128, shuffle=False, num_workers=1, pin_memory=True)
    xTest, yTest = DMP.DataLoaderToTensor(valLoader)
    yTarget = DMP.GenerateTargetsLabelRandomly(yTest, 10).type(torch.LongTensor).cuda()
    #Run the attack
    criterion = foolbox.criteria.TargetedMisclassification(yTarget[0:attackSampleNum])
    attack = LinfPGD(steps=10, random_start=True)
    epsilons = [0.05]
    raw_advs, clipped_advs, success = attack(fmodel, xTest[0:attackSampleNum].cuda(), criterion=criterion, epsilons=epsilons)
    clean_acc = accuracy(fmodel, raw_advs[0], yTarget[0:attackSampleNum])
    print(f"clean accuracy:  {clean_acc * 100:.1f} %")

Thank you and my apologies if this is not the right place to be posting this.

**Comment by caposerenity** [comment]
have you solved this problem?
